# **DATA CLEANING & PREPROCESSING**

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [6]:
df = pd.read_csv(r"C:\\Users\\\HP\\Downloads\\Amazon.csv")
df.head()

,OrderID,OrderDate,CustomerID,CustomerName,ProductID,ProductName,Category,Brand,Quantity,UnitPrice,...,PaymentMethod,OrderStatus,City,State,Country,SellerID,Year,Month,Day,DayOfWeek
0,ORD0000001,2023-01-31,CUST001504,Vihaan Sharma,P00014,Drone Mini,Books,BrightLux,3,106.59,...,Debit Card,Delivered,Washington,DC,India,SELL01967,2023,1,31,Tuesday
1,ORD0000002,2023-12-30,CUST000178,Pooja Kumar,P00040,Microphone,Home & Kitchen,UrbanStyle,1,251.37,...,Amazon Pay,Delivered,Fort Worth,TX,United States,SELL01298,2023,12,30,Saturday
2,ORD0000003,2022-05-10,CUST047516,Sneha Singh,P00044,Power Bank 20000mAh,Clothing,UrbanStyle,3,35.03,...,Debit Card,Delivered,Austin,TX,United States,SELL00908,2022,5,10,Tuesday
3,ORD0000004,2023-07-18,CUST030059,Vihaan Reddy,P00041,Webcam Full HD,Home & Kitchen,Zenith,5,33.58,...,Cash on Delivery,Delivered,Charlotte,NC,India,SELL01164,2023,7,18,Tuesday
4,ORD0000005,2023-02-04,CUST048677,Aditya Kapoor,P00029,T-Shirt,Clothing,KiddoFun,2,515.64,...,Credit Card,Cancelled,San Antonio,TX,Canada,SELL01411,2023,2,4,Saturday


In [7]:
df.shape

(98640, 24)

In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 98640 entries, 0 to 98639
Data columns (total 24 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   OrderID        98640 non-null  str    
 1   OrderDate      98640 non-null  str    
 2   CustomerID     98640 non-null  str    
 3   CustomerName   98640 non-null  str    
 4   ProductID      98640 non-null  str    
 5   ProductName    98640 non-null  str    
 6   Category       98640 non-null  str    
 7   Brand          98640 non-null  str    
 8   Quantity       98640 non-null  int64  
 9   UnitPrice      98640 non-null  float64
 10  Discount       98640 non-null  float64
 11  Tax            98640 non-null  float64
 12  ShippingCost   98640 non-null  float64
 13  TotalAmount    98640 non-null  float64
 14  PaymentMethod  98640 non-null  str    
 15  OrderStatus    98640 non-null  str    
 16  City           98640 non-null  str    
 17  State          98640 non-null  str    
 18  Country        98

In [9]:
df.isnull().sum()

OrderID          0
OrderDate        0
CustomerID       0
CustomerName     0
ProductID        0
ProductName      0
Category         0
Brand            0
Quantity         0
UnitPrice        0
Discount         0
Tax              0
ShippingCost     0
TotalAmount      0
PaymentMethod    0
OrderStatus      0
City             0
State            0
Country          0
SellerID         0
Year             0
Month            0
Day              0
DayOfWeek        0
dtype: int64

# **Missing Value Treatment**
No missing values were found in the dataset. Therefore, no imputation techniques were required.

In [10]:
df.duplicated().sum()

np.int64(0)

# **Duplicate Handling**
No duplicate records were found in the dataset. Hence, no duplicate removal was necessary.

In [11]:
df["OrderDate"] = pd.to_datetime(df["OrderDate"], dayfirst=True)

C:\Users\HP\AppData\Local\Temp\ipykernel_17420\1942507084.py:1: UserWarning: Parsing dates in %Y-%m-%d format when dayfirst=True was specified. Pass `dayfirst=False` or specify a format to silence this warning.
  df["OrderDate"] = pd.to_datetime(df["OrderDate"], dayfirst=True)


# **Data Type Conversion**
The OrderDate column was converted from string to datetime format to enable time-based analysis.

In [12]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 98640 entries, 0 to 98639
Data columns (total 24 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   OrderID        98640 non-null  str           
 1   OrderDate      98640 non-null  datetime64[us]
 2   CustomerID     98640 non-null  str           
 3   CustomerName   98640 non-null  str           
 4   ProductID      98640 non-null  str           
 5   ProductName    98640 non-null  str           
 6   Category       98640 non-null  str           
 7   Brand          98640 non-null  str           
 8   Quantity       98640 non-null  int64         
 9   UnitPrice      98640 non-null  float64       
 10  Discount       98640 non-null  float64       
 11  Tax            98640 non-null  float64       
 12  ShippingCost   98640 non-null  float64       
 13  TotalAmount    98640 non-null  float64       
 14  PaymentMethod  98640 non-null  str           
 15  OrderStatus    98640 non-null 

In [13]:
df["Category"] = df["Category"].str.strip()
df["Brand"] = df["Brand"].str.strip()
df["PaymentMethod"] = df["PaymentMethod"].str.strip()
df["OrderStatus"] = df["OrderStatus"].str.strip()
df["City"] = df["City"].str.strip()
df["State"] = df["State"].str.strip()
df["Country"] = df["Country"].str.strip()

# **Categorical Data Cleaning**
Leading and trailing spaces were removed from categorical columns such as Category, Brand, PaymentMethod, OrderStatus, City, State, and Country to ensure consistency.

In [14]:
Q1 = df["TotalAmount"].quantile(0.25)
Q3 = df["TotalAmount"].quantile(0.75)

IQR = Q3 - Q1

lower = Q1 - 1.5 * IQR
upper = Q3 + 1.5 * IQR

outliers = df[(df["TotalAmount"] < lower) | (df["TotalAmount"] > upper)]

outliers.shape

(437, 24)

# **Outlier Detection and Treatment**
Outliers in the TotalAmount column were identified using the IQR method.

Records outside the acceptable range were removed to improve data quality and ensure accurate analysis.

In [15]:
df = df[(df["TotalAmount"] >= lower) & (df["TotalAmount"] <= upper)]

In [16]:
df.shape

(98203, 24)

# **Final Dataset After Cleaning**
After removing outliers, the dataset size was reduced from 100,000 rows to 98,640 rows, ensuring better data quality.

In [17]:
df.to_csv(r"C:\Users\HP\Downloads\Eda Project\Data\processed\cleaned.csv", index=False)